ergani/models.py

```diff
@@ -0,0 +1,217 @@
+    business_branch_number: str
+    sepe_service_code: str
+    business_primary_activity_code: str
+    business_branch_activity_code: str
+    kallikratis_municipal_code: str
+    legal_representative_tax_identification_number: str
+    employee_overtimes: List[Overtime] = field(default_factory=list)
+    related_protocol_id: Optional[str] = ""
+    related_protocol_date: Optional[date] = None
+    employer_organization: Optional[str] = ""
+    business_secondary_activity_code_1: Optional[str] = ""
+    business_secondary_activity_code_2: Optional[str] = ""
+    business_secondary_activity_code_3: Optional[str] = ""
+    business_secondary_activity_code_4: Optional[str] = ""
+    comments: Optional[str] = ""
+
+    def serialize(self):
+        return {
+            "f_aa_pararthmatos": self.business_branch_number,
+            "f_rel_protocol": self.related_protocol_id,
+            "f_rel_date": format_date(self.related_protocol_date),
+            "f_ypiresia_sepe": self.sepe_service_code,
+            "f_ergodotikh_organwsh": self.employer_organization,
+            "f_kad_kyria": self.business_primary_activity_code,
+            "f_kad_deyt_1": self.business_secondary_activity_code_1,
+            "f_kad_deyt_2": self.business_secondary_activity_code_2,
+            "f_kad_deyt_3": self.business_secondary_activity_code_3,
+            "f_kad_deyt_4": self.business_secondary_activity_code_4,
+            "f_kad_pararthmatos": self.business_branch_activity_code,
+            "f_kallikratis_pararthmatos": self.kallikratis_municipal_code,
+            "f_comments": self.comments,
+            "f_afm_proswpoy": self.legal_representative_tax_identification_number,
+            "Ergazomenoi": {
+                "OvertimeErgazomenosDate": [
+                    overtime.serialize() for overtime in self.employee_overtimes
+                ]
+            },
+        }
+
+
+@dataclass
+class WorkdayDetails:
+    work_type: ScheduleWorkType
+    start_time: time
+    end_time: time
+
+    def serialize(self):
+        return {
+            "f_type": get_ergani_work_type(self.work_type),
+            "f_from": format_time(self.start_time),
+            "f_to": format_time(self.end_time),
+        }
+
+
+@dataclass
+class EmployeeSchedule:
+    employee_tax_identification_number: str
+    employee_last_name: str
+    employee_first_name: str
+    schedule_date: date
+    workday_details: List[WorkdayDetails] = field(default_factory=list)
+
+    def get_date_field(self, is_weekly: bool):
+        return "f_day" if is_weekly else "f_date"
+
+    def serialize(self, is_weekly: bool):
+        date_field = self.get_date_field(is_weekly)
+        return {
+            "f_afm": self.employee_tax_identification_number,
+            "f_eponymo": self.employee_last_name,
+            "f_onoma": self.employee_first_name,
+            date_field: get_day_of_week(self.schedule_date)
+            if is_weekly
+            else format_date(self.schedule_date),
```

Comment:
Instead of doing this, I suggest creating two different classes for employee WTO, according to use case; weekly or daily.

Reply:
@parisk That would mean that we would have to decouple `CompanyDailyShedule` and `CompanyWeeklySchedule`, which would lead to quite enough of code duplication.

Is this what you had in mind?

Reply:
Can we use inheritance to avoid code duplication? Also, it's a good idea to have the same signature for `.serialize()` everywhere, so we can also build abstract mechanisms around it, like DRF does on the server-side.